### PR TITLE
feat: migrate less to token for Alert

### PR DESCRIPTION
**Pull request by linhf123**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- #41884 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      migrate less to token for Alert      |
| 🇨🇳 Chinese |    迁移 `Alert` 组件的 `less` 变量到 `token`     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 4e0d10e</samp>

This pull request refactors the alert component style to use a new interface and a custom token generator function. It also updates the documentation for migrating less variables to component tokens in both English and Chinese, and removes the unnecessary documentation for some components.

### <samp>🤖 Generated by Copilot at 4e0d10e</samp>

*  Add alert component variables and tokens to support cssinjs migration ([link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-bbbc7b2eb93de110b862d8d30e7a416dab4e3868f9229c53edabbee04363f6baL3-R14), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-bbbc7b2eb93de110b862d8d30e7a416dab4e3868f9229c53edabbee04363f6baL40-R44), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-bbbc7b2eb93de110b862d8d30e7a416dab4e3868f9229c53edabbee04363f6baL52-R56), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-bbbc7b2eb93de110b862d8d30e7a416dab4e3868f9229c53edabbee04363f6baL231-R245), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL14-R22), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L14-R22))
*  Remove unused or redundant component variables from the migration documentation (`docs/react/migrate-less-variables.en-US.md` and `docs/react/migrate-less-variables.zh-CN.md`) ([link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL42), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL56), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL137), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL148), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL157), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL167), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L42), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L56), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L137), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L148), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L157), [link](https://github.com/ant-design/ant-design/pull/42142/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L167))

### CODE REVIEW

File: components/alert/style/index.ts [hunk 1]
```diff
@@ -228,13 +232,14 @@ export const genAlertStyle: GenerateStyle<AlertToken> = (token: AlertToken): CSS
...
-  const { fontSizeHeading3 } = token;
-
-  const alertToken = mergeToken<AlertToken>(token, {
-    alertIconSizeLG: fontSizeHeading3,
+export default genComponentStyleHook(
+  'Alert',
+  (token) => {
+    return [genAlertStyle(token)];
+  },
+  (token) => ({
```
Comment (any1024) [inline components/alert/style/index.ts:240]: 看了其他的 `pr`，`componentToken` 被移动到了第二个 `fn` 其实没太懂这里移动的意义

Comment (any1024) [reply components/alert/style/index.ts:238]: 看了下 `ts `定义应该是吐出每个组件自己的 `ComponentToken` ？

Comment (MadCcc) [reply components/alert/style/index.ts:238]: 是的，第二个 fn 用于初始化，在第一个 fn 的参数里 token 就已经包含了已经被初始化的 Component Token

Comment (any1024) [reply components/alert/style/index.ts:238]: > 是的，第二个 fn 用于初始化，在第一个 fn 的参数里 token 就已经包含了已经被初始化的 Component Token

好的，有空帮忙 cr 下

File: docs/react/migrate-less-variables.zh-CN.md [hunk 2]
```diff
@@ -11,7 +11,14 @@ title: Less 变量迁移 Design Token
...
 
-<!-- ### Alert 警告提示 -->
+## Alert 警告提示
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@alert-with-description-icon-size` | `alertIconSizeLG` | - |
+| `@alert-padding-horizontal` | `alertPaddingHorizontal` | - |
+| `@alert-padding-vertical` | `alertPaddingVertical` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:21]:
```suggestion
| `@alert-padding-horizontal` | - | 可由 `className` 和 `style` 直接修改 |
| `@alert-padding-vertical` | - | 可由 `className` 和 `style` 直接修改 |
```
这两个就不加了，直接用 className 和 style 覆盖就可以

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -11,7 +11,14 @@ title: Less 变量迁移 Design Token
 
 ## 组件变量
 
-<!-- ### Alert 警告提示 -->
+## Alert 警告提示
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@alert-with-description-icon-size` | `alertIconSizeLG` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:19]: 这应该不等价的

**Comment by MadCcc** [comment]
image
Alert 的 token 还是很多的，不管是能迁移还是不能迁移的都要写在文档里哈

**Comment by MadCcc** [comment]
命名方式参考 https://github.com/ant-design/ant-design/wiki/API-Naming-rules#component-token
基本都是把前缀摘了就行。

File: docs/react/migrate-less-variables.zh-CN.md (hunk 2 above)
Comment (any1024) [reply docs/react/migrate-less-variables.zh-CN.md:21]: > 命名方式参考 https://github.com/ant-design/ant-design/wiki/API-Naming-rules#component-token 基本都是把前缀摘了就行。

好的，我在参考下其他 pr 的实现改一下，当时以为只要将 Alert 几个变量列一下就行

File: docs/react/migrate-less-variables.zh-CN.md (hunk 3 above)
Comment (any1024) [reply docs/react/migrate-less-variables.zh-CN.md:48]: 看起来是等价的
image
image

**Review by MadCcc — APPROVED** [review]

**Pull request merged**